_posts/2024-12-26-issues-with-microros.md

```diff
@@ -0,0 +1,230 @@
+---
+title: "Less is More: Why We Abandoned Micro-ROS for UART"
+date: 2024-11-29
+categories: [Firmware]
+tags: [ros, control, motors, micro-ros, communication]     # TAG names should always be lowercase
+author: sis
+---
+
+
+If you've worked on robotics projects, chances are someone has mentioned ROS (Robot Operating System). And if you're anything like me, you’re always eager to try out shiny new tools.
+
+When we started working on our Two Wheel Inverted Pendulum (TWIP) project, I was determined to use ROS—mainly because it's a that counteless robotics videos swear by. 
+Augusto saw things differenlty. 
+He argued that *"ROS is massive, complex, requires a steep learning curve, and is most definitely overkill for what we're doing."* 
+His point was valid, but I wanted to explore what ROS could offer. Of course, I didn’t say that.
+Instead, I just said that communication shouldn’t be our main focus—so why not let ROS handle it? 
+
+Since then, our relationship with the ROS ecosystem has been, to put it mildly, turbulent.There is much to cover about the subject. This blog explores a small part of the story, mainly, using Micro ROS. 
```

Comment:
```suggestion
Since then, our relationship with the ROS ecosystem has been, to put it mildly, turbulent. There is much to cover about the subject. This blog explores a small part of the story, by mainly using micro-ROS. 
```